src/core/resource-container-helper.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {rethrowAsync} from './rethrow-async';
+
+const AMP_CLASS = 'i-amphtml-element';
+const DEEP = true;
+
+const load = (element) => element.ensureLoaded();
+const pause = (element) => element.pause();
+const unmount = (element) => element.unmount();
+
+/**
+ * Ensure all elements within this container are scheduled to load.
+ *
+ * @param {!Element|!Array<!Element>} containerOrContainers
+ * @param {boolean=} includeSelf
+ */
+export function loadAll(containerOrContainers, includeSelf = true) {
+  forAllWithin(containerOrContainers, includeSelf, !DEEP, load);
```

Comment:
Why is this one shallow, while the rest are deep?

Reply:
It's not very symmetric. When a component is paused or unmounted, all its children should also be paused/unmounted. It's safest to just do it immediately. With Owners we relied on a component to do so, but we've learned that it was very error-prone. We are not saving/losing anything from doing it right away here.

With loading the situation is different. If a parent is loaded, it doesn't mean that all children must load at the same time - that's almost never the case.